Sources/Sandbox.Game/Game/Entities/MyEntity.cs

```diff
@@ -463,6 +470,26 @@ void Components_ComponentRemoved(Type t, Sandbox.Common.Components.MyComponentBa
             }
         }
 
+        public void AssignGamelogicFromHashSet(HashSet<Type> items)
+        {
+            bool first = true;
+            foreach (Type logicType in items)
+            {
+                if (first)
+                {
+                    GameLogic = (MyGameLogicComponent)Activator.CreateInstance(logicType);
```

Comment:
Order does not matter here.  This is just to fill GameLogic first before filling the set.